ContextualPopupDecorator/tests/ContextualPopupDecorator-specs.js

```diff
@@ -122,4 +197,298 @@ describe('ContextualPopupDecorator Specs', () => {
+					Hello
+				</ContextualButton>
+			</Root>
+		);
+		const contextualPopup = screen.getByRole('alert');
+
+		const expected = 'left top';
+		const actual = contextualPopup.children.item(0);
+
+		expect(actual).toHaveClass(expected);
+	});
+
+	test('should set pointerMode to be false when directional key is pressed', () => {
+		const Root = FloatingLayerDecorator('div');
+		render(
+			<Root>
+				<ContextualButton open popupComponent={() => <div><Button>first</Button><Button>second</Button></div>}>
+					Hello
+				</ContextualButton>
+			</Root>
+		);
+		const contextualButton = screen.getByRole('alert');
+
+		leftKeyDown(contextualButton);
+
+		const expected = false;
+		const pointerMode = getPointerMode();
+
+		expect(pointerMode).toBe(expected);
+	});
+
+	test('should close popup if prop \'open\' is omitted on rerender', () => {
+		const Root = FloatingLayerDecorator('div');
+		const {rerender} = render(
+			<Root>
+				<ContextualButton open popupComponent={() => <div><Button>first</Button><Button>second</Button></div>}>
+					Hello
+				</ContextualButton>
+			</Root>
+		);
+		const firstRender = screen.getAllByRole('button').length;
+		const expectedFirst = 3;
+
+		rerender(
+			<Root>
+				<ContextualButton popupComponent={() => <div><Button>first</Button><Button>second</Button></div>}>
+					Hello
+				</ContextualButton>
+			</Root>
+		);
+		const secondRender = screen.getAllByRole('button').length;
+		const expectedSecond = 1;
+
+		expect(firstRender).toBe(expectedFirst);
+		expect(secondRender).toBe(expectedSecond);
+	});
+
+	test('should close popup if prop \'open\' is added on rerender', () => {
+		const Root = FloatingLayerDecorator('div');
+		const {rerender} = render(
+			<Root>
+				<ContextualButton popupComponent={() => <div><Button>first</Button><Button>second</Button></div>}>
+					Hello
+				</ContextualButton>
+			</Root>
+		);
+		const firstRender = screen.getAllByRole('button').length;
+		const expectedFirst = 1;
+
+		rerender(
+			<Root>
+				<ContextualButton open popupComponent={() => <div><Button>first</Button><Button>second</Button></div>}>
+					Hello
+				</ContextualButton>
+			</Root>
+		);
+		const secondRender = screen.getAllByRole('button').length;
+		const expectedSecond = 3;
+
+		expect(firstRender).toBe(expectedFirst);
+		expect(secondRender).toBe(expectedSecond);
+	});
+
+	test('should capture \'onKeyDown\' event from outside of popup', () => {
+		const handleOnKeyDown = jest.fn();
+		const Root = FloatingLayerDecorator('div');
+		render(
+			<Root>
+				<ContextualButton onKeyDown={handleOnKeyDown} open popupComponent={() => <div><Button>first</Button><Button>second</Button></div>}>
+					Hello
+				</ContextualButton>
+			</Root>
+		);
+
+		const button = screen.getAllByRole('button')[0];
+		leftKeyDown(button);
+
+		expect(handleOnKeyDown).toHaveBeenCalled();
+	});
+
+	test('should render popup if \'scrimType\' has value \'holepunch\'', () => {
```

Comment:
@alexandrumorariu , I 'm not sure about this test case purpose. I think whether scrimType is holepunch or not, popup can be rendered. Could you please check this TC to remove or fix it?
Other changes looks good to me.

Reply:
We need this TC to increase the total coverage, I will find a better description/purpuse to render a component with the scrimType holepunch.